x-pack/plugins/security_solution/common/endpoint/utils/fetch_fleet_version.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Fetches the latest version of the Elastic Agent available for download
+ * @param kbnClient
+ */
+import type { KbnClient } from '@kbn/test';
+import { AGENT_API_ROUTES } from '@kbn/fleet-plugin/common';
+import { catchAxiosErrorFormatAndThrow } from '../format_axios_error';
+
+export const fetchFleetLatestAvailableAgentVersion = async (
+  kbnClient: KbnClient
+): Promise<string> => {
+  return kbnClient
+    .request<{ items: string[] }>({
```

Comment:
Does fleet have a Type for this API response? if so, I suggest you use it here

Reply:
typed as `string[]` fleet api side. 

https://github.com/elastic/kibana/blob/6b29552382cb84f908f02d5dc605b150d1ebbde5/x-pack/plugins/fleet/server/services/agents/versions.ts#L94

![Screenshot 2024-08-05 at 15 08 37](https://github.com/user-attachments/assets/aa25fba8-52d1-4e19-b860-e21b9db9421d)

Reply:
That seemed strange to me since Fleet normally types all of their interfaces, so looked it up quickly and there is one. Just FYI: when you are looking for Types for their APIs, you look under `fleet/common/type/**`.

here is the type for this API: https://github.com/elastic/kibana/blob/6b29552382cb84f908f02d5dc605b150d1ebbde5/x-pack/plugins/fleet/common/types/rest_spec/agent.ts#L239

Reply:
You're right, I should have checked more thoroughly :). Thanks!